ref(native-stack-trace-v2): Add native stack trace new content design [NATIVE-343]

priscilawebdev opened the pull request:
- [x] Update the `render` logic of the ContentV3 to only be displayed when the platform is of type native
- [x] Update the ContentV3 file to render the new nativeFrame file + CSS updates
- [x] Create a new file called nativeFrame,  which renders the new design styled defined in the figma file https://www.figma.com/file/nZJDiuOgaLlU8pCSB1RjD4/Native-Stack-Trace?node-id=247%3A7318

**Before:**

_Preview_
![image](https://user-images.githubusercontent.com/29228205/151398873-f477fdf3-cb50-4f84-8ead-e7dee3a6146f.png)


_Issue Details_
![image](https://user-images.githubusercontent.com/29228205/151398494-b14200b7-b8fc-4371-aadb-d094ce857637.png)


**After:**

_Preview_

![image](https://user-images.githubusercontent.com/29228205/152135957-0bb31631-db46-41c9-9c16-70de24b151ee.png)



_Issue Details_

![image](https://user-images.githubusercontent.com/29228205/152135883-c53e3013-334e-4db0-b6ec-dc0f0339db05.png)



_Fix Empty Message Icon Size_

**Before:**
![image](https://user-images.githubusercontent.com/29228205/151780092-e7774ef5-16d8-4177-9c15-20388124299b.png)



**After:**
![image](https://user-images.githubusercontent.com/29228205/151780041-ec8b512d-f369-4afb-9fa9-87f368ca6d58.png)


ps: Every change is behind a feature flag

TODO: 
- [ ] Tests will be created in a follow-up PR after everything is approved and all feedback is applied
- [ ] Update how the `renderOmittedFrames` is rendered

priscilawebdev committed:
ref(native-stack-trace-v2): Add native stack trace new content design

wip

wip

mobile style

cleanup

wip

wip

wip

minor adjusts

wip

priscilawebdev committed:
remove display

priscilawebdev committed:
Merge branch 'master' into ref/native-stack-trace-new-content

Jesse-Box commented:
Heyy!

So I looked at this and here are some things I would like tweaked.
- The grid-column-template(**large viewport only**): I think it will be best if we used fixed units for most row columns. `grid-template-columns: 24px 132px 116px 24px 1fr 24px;` worked well in most cases for me. I love the idea of a more flexible stack trace, but in practice the user will get less line length for the function & file column.
- Using `Roboto Mono` for the instruction address
- `gray300` for "Crashed, Called from" text as well as the file name
- The `AddressCell` has a `padding-left 8px;`which isn't needed (Not sure if this used for smaller viewports?)
- When the function and file name collapse onto 3 lines, the package name has some weird spacing issue that needs fixing. (See Below)
<img width="921" alt="Screen Shot 2022-01-28 at 16 20 47" src="https://user-images.githubusercontent.com/6508377/151573396-905a7a8d-fd69-4821-af33-61328dc4884c.png">

Excellent stuff @priscilawebdev !!

priscilawebdev committed:
Merge branch 'master' into ref/native-stack-trace-new-content

priscilawebdev committed:
cr feedback

priscilawebdev committed:
update empty message to have an larger icon

priscilawebdev commented:
Thanks for the feedback @Jesse-Box 

. I updated the grid layout values but in the address column instead of `116px` I added `138px` as I noticed that some addresses were too long for the column and were breaking into multiple lines
. The Robot Font was there, but inside of a breakpoint. I've fixed it..thanks for pointing this out
. "Crashed, Called from"  have now the `gray300` color
. The `AddressCell` had 8px spacing on the left side because I thought it looked better with more spacing, but I've already removed it
. I updated the code to display less spacing. It looks like below:


![image](https://user-images.githubusercontent.com/29228205/151790400-f4d4b810-2662-4257-a7f7-f26d02d865de.png)

Jesse-Box commented:
Hey @priscilawebdev looks great!

Only one feedback point to give

Can you make the file paths next to the function (seen below) the same text colour as "Crashed in non-app" AND give a dotted line to indicate the tooltip behaviour.
<img width="241" alt="Screen Shot 2022-02-01 at 10 29 52" src="https://user-images.githubusercontent.com/6508377/151943898-fa263269-6877-40d2-8bdf-5cc305a2a08b.png">
e

matejminar commented on static/app/components/events/interfaces/nativeFrame.tsx:
```diff
@@ -0,0 +1,460 @@
+`;
+
+const Package = styled('span')<{color?: Color}>`
+  border-bottom: 1px dashed ${p => p.theme.gray200};
```
```suggestion
  border-bottom: 1px dashed ${p => p.theme.border};
```

matejminar reviewed (approved):
The code looks good to me.

One observation:
Clicking on `Go to images loaded` takes you to the filtered down images loaded section. This section is collapsed by default nowadays. Should we consider opening it once the button is clicked?

priscilawebdev committed:
Update static/app/components/events/interfaces/nativeFrame.tsx

Co-authored-by: Matej Minar <matej.minar@sentry.io>

priscilawebdev commented:
@matejminar thanks!

Yes I will fix those regressions in a follow up PR

priscilawebdev committed:
Merge branch 'master' into ref/native-stack-trace-new-content

priscilawebdev committed:
Merge branch 'ref/native-stack-trace-new-content' of https://github.com/getsentry/sentry into ref/native-stack-trace-new-content

priscilawebdev committed:
cr feedback

priscilawebdev commented:
@Jesse-Box Thanks for the Feedback. I've updated the screenshots

priscilawebdev committed:
Merge branch 'master' into ref/native-stack-trace-new-content

priscilawebdev committed:
fix tests
